modules/core/src/main/java/org/apache/synapse/aspects/flow/statistics/data/raw/StatisticsLog.java

```diff
@@ -163,6 +164,11 @@ public class StatisticsLog {
 	 */
 	private String statusDescription;
 
+	/**
+	 * MessageContext of the component.
+	 */
+	private MessageContext messageContext;
```

Comment:
is this StatisticsLog instance created per mediation ?

Reply:
Yes. Created mostly in the analytics reporting thread

Reply:
as discussed offline, let's try to remove this message context, as it might cause memory issues in future if not allowed to be properly GCed.

Reply:
Done!